ui/raidboss/data/05-shb/trial/diamond_weapon.js

```diff
@@ -0,0 +1,147 @@
+import ZoneId from '../../../../../resources/zone_id';
+import NetRegexes from '../../../../../resources/netregexes';
+import Conditions from '../../../../../resources/conditions';
+import { Responses } from '../../../../../resources/responses';
+
+const sharedOutputStrings = {
+  teleportEast: {
+    en: 'Teleport to east platform',
+  },
+  teleportWest: {
+    en: 'Teleport to west platform',
+  },
+};
+
+export default {
+  zoneId: ZoneId.TheCloudDeck,
+  timelineFile: 'diamond_weapon.txt',
+  triggers: [
+    {
+      id: 'Diamond Diamond Rain',
+      netRegex: NetRegexes.startsUsing({ source: 'The Diamond Weapon', id: '5FA7', capture: false }),
+      condition: Conditions.caresAboutAOE(),
+      durationSeconds: 4,
+      response: Responses.aoe(),
+    },
+    {
+      id: 'Diamond Claw Swipe East',
+      netRegex: NetRegexes.startsUsing({ source: 'The Diamond Weapon', id: '5F9A', capture: false }),
+      delaySeconds: 2,
```

Comment:
Could we write this as `(data, matches) => parseFloat(matches.duration) - DELAY`? This way it's easier to see exactly how long before the end of the cast we want the user to be notified, as opposed to approximating it with an arbitrary number.

Reply:
I can; in this case it would be `(data, matches) => parseFloat(matches.duration) - 2,` though (if that makes a difference).
Alternatively, these don't really need a delay, or also don't really need to be called this early. 
I decided to go the middle route between the two and call it halfway through the castbar and extend the duration telling you what to do.

Reply:
That's fair. I don't really care about *whether* there's a delay, honestly, just that if there *is* we display it as a function so we know how much time we're giving before the cast bar completes.

Reply:
My general rule of thumb is to call things as early as possible.  The only exceptions are really knockbacks (call 5 seconds beforehand) or maybe something like "move around"/"stop" where you want to call it closer to the point where somebody has to take (or not take) action vs having them stand around and do nothing for a bit.  Something like "move to the other side" should happen as soon as possible, imo.